classes/Cart.php

```diff
@@ -755,6 +755,15 @@ public function getProducts($refresh = false, $id_product = false, $id_country =
                     $reduction_type_row = ['reduction_type' => 0];
                 }
 
+                if (Pack::isPack($product['id_product'])) {
+                    /*
+                    * In case of packs, we need to properly calculate the quantity in stock. We can't just use the quantity from the query, because stocks can be set to use the quantity of the products in them. The quantity in stock_available then has no meaning and could be always zero.
+                    *
+                    * When calling Pack::getQuantity here, you MUST use null for $cart parameter. Otherwise it will subtract the quantity that is already in the cart. Basically resulting in a nonsense - half of quantity you have. We need the REAL quantity.
+                    */
+                    $row['quantity_available'] = Pack::getQuantity((int) $product['id_product'], (int) $product['id_product_attribute'], true, null, (int) $product['id_customization']);
```

Comment:
I'm not sure if setting `$row['quantity_available']` will make any difference.

Reply:
`$row` doesn't exist

Reply:
@kpodemski, @Hlavtox
sorry, you're right, I copied and pasted and didn't notice the error.

I made the correction.